docs/how-to/create-chiselled-docker-image.md

```diff
@@ -0,0 +1,181 @@
+# Create a chiselled Docker image
+
+Since Chisel can produce a minimal root file system, using it in Docker images
+has lots of value. This guide shows how to create a chiselled Docker image.
+
+A _chiselled_ image is one which only contains files installed by the slices and
+does not contain any base images such as the ubuntu image.
+
+
+## Design the image
+
+It's a good idea to think first about what our chiselled image should be able to
+do. Let's consider a Python 3 image, which contains a Python 3 interpreter and
+is able to run Python 3 scripts and commands.
+
+Goals:
+
+- The [image entrypoint] must be `python3`.
+- The image must be chiselled i.e. there must not be anything else.
+- The image must be able to run Python 3 scripts.
+- The image must be minimal.
+
+To build this image, we will use [multi-stage docker builds].
+
+In earlier stage(s),
+
+1. We will install Chisel and other dependencies that we may
+need.
+2. We will install the slices using Chisel and prepare a root file system.
+
+Finally, we will copy the root file system to the `/` directory in the final
+stage. The final stage base image will be [`scratch`], i.e. nothing. Thus, the
+image only contains the root file system installed by Chisel and nothing else.
+
+
+## Install Chisel and dependencies
+
+To make building easier and configurable, let's first define some variables that
+we can re-use.
+
+```docker
+ARG UBUNTU_RELEASE=24.04
+ARG USER=app UID=101 GROUP=app GID=101
+ARG CHISEL_VERSION=1.1.0
+```
+
+Let's now initialize a new build stage where we install Chisel and dependencies
+and prepare the root file system.
+
+```docker
+FROM ubuntu:$UBUNTU_RELEASE AS builder
+ARG USER UID GROUP GID TARGETARCH UBUNTU_RELEASE CHISEL_VERSION
+SHELL ["/bin/bash", "-oeux", "pipefail", "-c"]
+```
+
+We will need the `ca-certificates` package to fetch files. Let's install that.
+
+```docker
+RUN apt-get update \
+    && DEBIAN_FRONTEND=noninteractive apt-get install -y ca-certificates \
+    && apt-get clean -y \
+    && rm -rf /var/lib/apt/lists/*
```

Comment:
I'd do some more research about this. I know it's common practice, but I always frowned upon this practice within a multi stage build, cause it only slows down the build, and it's irrelevant (since this is just a build stage). 

If seen as a best practice, then keep it

Reply:
It's a best practice, indicated in the docker docs: https://docs.docker.com/build/building/best-practices/#apt-get

Reply:
right, I know, but that page isn't specific to multi-stage. One should definitely do this in the target stages. But in intermediary stages, what's the drawback? larger cache maybe...but additional instructions polluting the dockerfile and slowing down the build...anyway, it's a nitpick, forget about it